examples/multi.rs

```diff
@@ -0,0 +1,74 @@
+use std::env;
+use std::time::Duration;
+
+use anyhow::Result;
+
+use curl::easy::Easy;
+use curl::multi::{EasyHandle, Message, Multi};
+
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(10);
+
+fn perform_requests(projects: &Vec<String>) -> Result<Vec<Option<u32>>> {
+    fn create_request(multi: &Multi, url: &str, token: usize) -> Result<EasyHandle> {
+        let version = curl::Version::get();
+        let mut request = Easy::new();
+        request.url(url)?;
+        request.timeout(REQUEST_TIMEOUT)?;
+        request.useragent(&format!("curl/{}", version.version()))?;
+        request.write_function(|data| {
+            // You can save the data here.
+            Ok(data.len())
+        })?;
+
+        let mut handle = multi.add(request)?;
+        handle.set_token(token)?;
+        Ok(handle)
+    }
+
+    fn parse_response(handle: &mut EasyHandle, message: &Message) -> Option<u32> {
+        message.result()?.ok()?;
+        handle.response_code().ok()
+    }
+
+    let multi = Multi::new();
+    let mut handles: Vec<_> = projects
+        .iter()
+        .enumerate()
+        .map(|(token, url)| create_request(&multi, url, token).unwrap())
+        .collect();
+
+    let mut statuses: Vec<Option<u32>> = Vec::new();
+    statuses.resize(projects.len(), None);
+
+    let mut completed = 0;
+    while completed < handles.len() {
+        // Let curl multi perform as much transfer as possible on every
+        // attached "easy" handles.
+        multi.perform()?;
+
+        multi.messages(|message| {
+            let token = message.token().expect("failed to get the token");
+            assert!(token <= statuses.len());
+            statuses[token] = parse_response(&mut handles[token], &message);
+            completed += 1;
+        });
+
+        std::thread::sleep(Duration::from_millis(16));
+    }
+
+    Ok(statuses)
+}
+
+fn main() -> Result<()> {
+    let urls: Vec<String> = env::args().skip(1).collect();
+
+    curl::init();
```

Comment:
I think that this can be skipped in examples?

Reply:
I noticed that it's not necessary under specified scenario described in the doc.

I ended up adding it, because of the following quote:
> ...it is strongly recommended that you call this function manually as soon as your program starts.

I don't have a big issue either way.

Reply:
I think we may want to tone down the recommendation there, but for small examples it shouldn't be necessary (or it's a bug in the `curl` crate)

Reply:
You missed the entire sentence in your quote (emphasis mine):

> This should be sufficient for most applications and scenarios, **but in any other case**, it is strongly recommended that you call this function manually as soon as your program starts.

It is strongly recommended that you call it manually only in the scenarios that are not already covered. Perhaps it could be reworded to be more clear, but basically, "If your program _doesn't_ match these requirements, then you should call it manually."

Since the example is single-threaded and isn't targeting uncommon platforms it is not necessary to call.